### PR TITLE
Add ability to run tasks by category/group

### PR DESCRIPTION
**Pull request by scalvert**
This change expands the existing functionality of the command line options to include the ability to run tasks by either category or group.

Similar to the `task` option, you can now pass a `category` option (multiple times):

```shell
checkup --category 'best practices'
```

or a `group` option (multiple times)

```shell
checkup --group 'ember'
```

### CODE REVIEW

File: packages/cli/src/commands/run.ts [hunk 1]
```diff
@@ -77,6 +78,14 @@ export default class RunCommand extends BaseCommand {
       char: 'd',
       description: 'The path referring to the root directory that Checkup will run in',
     }),
+    category: flags.string({
+      description: 'Runs specific task specified by category. Can be used multiple times.',
```
Comment (carakessler) [inline packages/cli/src/commands/run.ts:82]: nit: should be `tasks`

File: packages/cli/src/commands/run.ts [hunk 2]
```diff
@@ -77,6 +78,14 @@ export default class RunCommand extends BaseCommand {
...
+      multiple: true,
+    }),
+    group: flags.string({
+      description: 'Runs specific task specified by group. Can be used multiple times.',
```
Comment (carakessler) [inline packages/cli/src/commands/run.ts:86]: samezies

File: packages/cli/src/commands/run.ts [hunk 3]
```diff
@@ -114,6 +123,18 @@ export default class RunCommand extends BaseCommand {
   checkupConfig!: CheckupConfig;
   cliModeEnabled: boolean = true;
 
+  get taskFilterType() {
+    if (this.runFlags.task !== undefined) {
+      return 'task';
+    } else if (this.runFlags.category !== undefined) {
```
Comment (carakessler) [inline packages/cli/src/commands/run.ts:129]: we should specify that groups and category flags cant be used together if this is going to be the business logic (FWIW i think its fine to have it be one or the other, would just want to make that clear in docs)

**Review by carakessler — APPROVED** [review]
Overall LGTM, just a few nits on documentation

File: packages/cli/src/commands/run.ts (hunk 1 above)
Comment (scalvert) [reply packages/cli/src/commands/run.ts:82]: Fixed!

File: packages/cli/src/commands/run.ts (hunk 2 above)
Comment (scalvert) [reply packages/cli/src/commands/run.ts:86]: Fixed!

File: packages/cli/src/commands/run.ts (hunk 3 above)
Comment (scalvert) [reply packages/cli/src/commands/run.ts:129]: This is a good call. I'll make sure it's called out in the docs when I update them.

**Pull request merged**